### ISSUE TITLE
Sheets doesn't load when self-hosting

### ISSUE THREAD

**Issue by thomasjsn** [xwiki-labs/cryptpad#518]
I am unable to load sheets on my self-hosted instance, https://pad.homelab.no/sheet

It just loads forever, and the following errors appair in the console:

```
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 2 utils.js:35:9
uncaught exception: EvalError: call to eval() blocked by CSP jquery.min.js:4:15966
EvalError: call to eval() blocked by CSP jquery.min.js:2:2640
uncaught exception: TypeError: window.DocsAPI is undefined sframe-boot2.js:45:9
TypeError: window.DocsAPI is undefined inner.js:1004:13
Content Security Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). jquery.min.js:2:2639
Content Security Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). a849e9ed-83d6-4f53-bfaa-116e0a5aa8fa:27:22
Compiling [/share/app-share.less] took 5ms LessLoader.js:161:25
creating favicon notify.js:70:17
```

* Windows 10
* Firefox 74.0
* Extensions: AdBlock, Privacy badger
* CryptPad v3.14.0

All other apps loads just fine, but not Sheets.

Related to #502?

**Comment by kitoy30** [comment]
Hi, you can look in your nginx configuration if you have comment this lines : 
if ($uri = "/pad/inner.html") { set $unsafe 1; }
if ($uri = "/sheet/inner.html") { set $unsafe 1; }

If you have make your installation with ansible, you can modify you config.js  like this: 

 padContentSecurity: [
      "default-src 'none'",
      "style-src 'unsafe-inline' ", + domain,
      "script-src 'self' ", + domain,
      "font-src 'self'", + domain,
      ]

The toolbar generate CSP errors otherwise

**Comment by thomasjsn** [comment]
> Hi, you can look in your nginx configuration if you have comment theses lines, just comment them :
> 
> if ($uri = "/sheet/inner.html") { set $unsafe 1; }

I don't have a nginx configuration since I am using Haproxy in front of CryptPad. What does that line do?

**Comment by kitoy30** [comment]
This lines add to script-src  'unsafe-inline' and  'unsafe-eval' 
It generates CSP bug.

**Comment by etsstiffler** [comment]
Hi,
i have the same error, but i'm using a apache sever. Is there some expample where i have to change these things?
I don't know if it'S related to this issue, but if i use the docker-compose file, cryptpad refuses the connection.

**Comment by ansuz** [comment]
The root cause of the issue is a trailing slash after the `httpSafeDomain` value in your config file. The client uses this value to construct URLs and produces some paths with two leading slashes. These incorrect paths then fail to match the pattern which sends the appropriate CSP headers, which breaks the applications which need special CSP rules to function (sheets and rich text).

There is a fix on our staging branch in which the server removes trailing slashes after loading the config file, but in the meantime you can just edit your config and restart your server. We plan to merge staging to master and release these changes as our next version next Tuesday (April 7th, 2020).

**Comment by ansuz** [comment]
A fix which causes the server to ignore trailing slashes in config files has been merged to master and tagged as a part of our [3.15.0 release](https://github.com/xwiki-labs/cryptpad/releases/tag/3.15.0).
